### ISSUE TITLE
VEVO bug (aka Video play error)

### ISSUE THREAD

**Issue by vicone666** [ram-on/SkyTube#7]
VEVO videos do not play

**Comment by husadd** [comment]
Try to change your "preferred resolution" to a lower one in settings, maybe your phone doesn't support 1080p or 720p so the video (which is available in 720p or 1080p) will keep loading. Or maybe your connection speed is slow so it takes long time to load it in a high resolution.
Hope that helps.

**Comment by ram-on** [comment]
As @husadd stated, try to lower the preferred resolution from the settings.  It is known that some devices do not support high resolution videos.

**Comment by vicone666** [comment]
Okay you guys were semi right, the problem only occurs if I do not pick any preferred resolution (aka default setting) otherwise vevo videos work, tested on 360p and 1080p. 

So I guess the current solution would be to have a default preference selection, dunno lets say default value 360p or 480p.

Keep up the good work tho

Device: Nexus 5, 6.0.1

**Comment by ram-on** [comment]
@vicone666 Which is the highest resolution that your device supports?

**Comment by vicone666** [comment]
1080p videos work flawlessly

**Comment by vicone666** [comment]
An error has occured when getting video id: j5-yKhDd64s

While having not having any option selected in the video quality preference.

**Comment by ram-on** [comment]
@vicone666 that error most probably means that, for example, the video is not available in your country.  Eventually, such error messages will be more user friendly and more descriptive.

**Comment by lfom** [comment]
I have the same problem: most music videos don't play (error getting video _code_), while they play on YT app (same Vevo URL). I have tried all res, no deal...

**Comment by ram-on** [comment]
@lfom pls specify the device name

**Comment by lfom** [comment]
LG L90 (w7dsn), running Marshmallow (CM13).

**Comment by lfom** [comment]
Exactly same error: S5 mini (SM-G800h) running stock 5.1.1 Lollipop. I am in Brazil but the same URL opens in YT and Vivo apps.

**Comment by ram-on** [comment]
@lfom Strange.  I have an S4 mini and it plays vevo videos.

**Comment by ram-on** [comment]
**To anyone who is encountering this error**:  it would be **really helpful** if you are attach the app's log file.

This can be done by:

1. Connecting your device to your computer
2. Running the app
3.  Execute:

```
adb logcat
```

or, ideally, by running this command:

```bash
adb logcat | grep `adb shell ps | grep free.rm.skytube | cut -c10-15`
````

This would be extremely helpful as I cannot reproduce this error on any or my devices.

**Comment by ram-on** [comment]
@r4nd0m123 thank you very much.  i'll soon analyse your ADB log.

**Comment by atomjack** [comment]
I think I tracked down the problem. Could some of you guys who had this problem post the youtube URL's of the videos you tried, so I can verify they all work with my fix?

**Comment by r4nd0m123** [comment]
@atomjack https://www.youtube.com/watch?v=hyPkpXZNbK0
If you need me to do any testing let me know.

**Comment by ram-on** [comment]
Fixed.  Refer to 39e6bbd178ac61d46d4fc3a730e7ffe8dc7b6e1a.  Thx to everyone who contributed.

**Comment by github-throwaway** [comment]
It seems that VEVO videos stopped working again. On my N9005 running Android 5.0 and the Skytube app 2.2 I get this error:

![screenshot_2017-02-03-14-52-59](https://cloud.githubusercontent.com/assets/25488086/22597606/95031090-ea30-11e6-9650-d13774e02231.png)
What's quite interesting is that it also applies to non VEVO channels. Trying to access this video (https://www.youtube.com/watch?v=fJBFvNiJ4Kw) I got the same error.

**Comment by ram-on** [comment]
The problem is that I can't replicate this bug whatsoever.  Anyone who is encountering with this bug and has some knowledge in Java is encouraged to contribute.  @atomjack perhaps?

**Comment by ram-on** [comment]
This should be resolved now due to d249e5f848b0338d549df580fb08ea3d69b0b987.
